lib/pbench/server/api/resources/query_apis/metadata_index/index_mappings.py

```diff
@@ -56,23 +66,41 @@ class IndexMappings(Resource):
     }
     """
 
-    def __init__(self, logger: Logger):
-        self.logger = logger
+    def __init__(self, config: PbenchServerConfig, logger: Logger):
+        super().__init__(
+            config,
+            logger,
+            Schema(
+                Parameter(
+                    "index_key",
+                    ParamType.KEYWORD,
+                    required=True,
+                    keywords=list(RunIdBase.ES_INTERNAL_INDEX_NAMES.keys()),
+                    uri_parameter=True,
+                )
+            ),
+        )
 
-    def get(self, index_key: AnyStr) -> Dict[AnyStr, Any]:
+    def _get(self, index_key: AnyStr, request: Request) -> Dict[AnyStr, Any]:
         """
-        Return mapping properties of the document specified by the index name
-        in the URI. For example, user can get the run document properties by
-        making a GET request on index/mappings/search. Similarly other index
-        documents can be fetched by making a GET request on appropriate index names.
+        Return mapping properties of the document specified by the index key
+        in the URI (supported index keys are defined in RunIdBase.ES_INTERNAL_INDEX_NAMES).
+        For example, user can get the run document properties by making a GET
+        request on index/mappings/search. Similarly other index documents can be
+        fetched by making a GET request on appropriate index names.
 
         We fetch the mapping by querying the template database. If the
         template is not found in the database NOT_FOUND error will be raised.
```

Comment:
If the _key_ is not found, it's a malformed request; if the corresponding template is not found, then it's some sort of internal configuration error.  (I think I would just delete the second sentence here and let the code speak for itself.)

Reply:
Yeah I will need to update the comment here.